src/plugin/ipc/file/filewrappers.cpp

```diff
@@ -371,7 +371,15 @@ static int _open_open64_work(int(*fn) (const char *path, int flags, ...),
     newpath = currPtsDevName;
   }
 
-  int fd = (*fn) (newpath, flags, mode);
+  dmtcp::string hook_path = "";
+  bool doSwap = pathvirt_get_physical_path ? pathvirt_get_physical_path(newpath, hook_path) : false;
+  int fd = -1;
+
+  /* hook_path was NULL, not swapping */
+  if (!doSwap)
+    fd = (*fn)(newpath, flags, mode);
+  else
+    fd = (*fn)(hook_path.c_str(), flags, mode);
 
   if (fd >= 0 && dmtcp_is_running_state()) {
     FileConnList::instance().processFileConnection(fd, newpath, flags, mode);
```

Comment:
@gc00 @rohgarg is always using `newpath` here a bug? if we have swapped, should we be using `hook_path.c_str()` in this `processFileConnection` call instead?